### PR TITLE
Add compatibility with Android Gradle Plugin 8.0

### PR DESCRIPTION
**Pull request by asaarnak**
Fixes issue #1036

### CODE REVIEW

**Comment by navaronbracke** [comment]
@mvanbeusekom Can we look into landing this PR? This is blocking adoption of Gradle 8 in other projects.

**Comment by penhorwood** [comment]
the gradle 8 issue is stopping me from using this project

File: permission_handler_android/android/build.gradle [hunk 1]
```diff
@@ -27,6 +27,10 @@ project.getTasks().withType(JavaCompile){
 apply plugin: 'com.android.library'
 
 android {
+    // Conditional for compatibility with AGP <4.2.
+    if (project.android.hasProperty("namespace")) {
+      namespace 'com.baseflow.permissionhandler'
+    }
```
[the remark this answers is not in the document]
Comment (asaarnak) [reply permission_handler_android/android/build.gradle:33]: https://github.com/flutter/flutter/issues/125181#issuecomment-1525784200

[the remark this answers is not in the document]
Comment (bartekpacia) [reply permission_handler_android/android/build.gradle:33]: This community is lucky to have people like the author of that comment :)

I wouldn't have that much mercy, haha.

**Review by mvanbeusekom — APPROVED** [review]
LGTM, thanks for creating this PR and helping out.

**Pull request merged**